src/main/java/com/google/devtools/build/lib/remote/RemoteSpawnCache.java

```diff
@@ -112,7 +115,7 @@
 
 
   @Override
-  public CacheHandle lookup(Spawn spawn, SpawnExecutionContext context)
+  public CacheHandle lookup(Spawn spawn, SpawnExecutionContext context, MetadataHandler metadataHandler)
```

Comment:
instead of passing in the MetadataHandler consider moving artifactOmitted to the MetadataProvider interface (from the MetadataHandler). This should really be as simple as moving the method between the interfaces.

Reply:
Okay!

Reply:
@buchgr I did that, but I noticed that a few classes are implementing `MetadataProvider`, and I made stubs in `artifactOmitted` in order to make sure everything is correct. Can you please take a look at that?